Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +121,62 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'The root directory cannot be created.');
+        Assert::startsWith($path, '/', 'The target path %s is not absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentNode = NodeHelper::createPath($this->getManager()->getPhpcrSession(), $resolvedPath);
+        if (!$parentNode instanceof NodeInterface) {
+            throw new InvalidArgumentException('No parent node created for '.$path);
+        }
```

Comment:
- NodeHelper will always return a PhpcrNode, we shouldn't bother testing that here.
- You are not retrieving/creating the parent node, but the node itself.

Also I think we `auto-creation` is another feature, we should initially only worry about retrieving the parent document:

```php
$parentPath = PathHelper::getParentPath($resolvedPath);
$parentDocument = $this->getManagr()->find(null, $parentPath);

if (null === $parentDocument) {
    throw new \Exception(sprintf('Cannot locate parent document at "%s"', $parentPath));
}
```